sunpy/net/tests/test_hek2vso.py

```diff
@@ -61,28 +64,98 @@ def test_translate_results_to_query():
 
     # Checking Instrument
-    assert vso_query[2].value == hek_query[0]['obs_instrument']
+    assert vso_query[2].value == hek_query[0]["obs_instrument"]
 
     # Checking Wavelength
-    assert vso_query[3].min == hek_query[0]['obs_meanwavel'] * u.Unit(hek_query[0]['obs_wavelunit'])
-    assert vso_query[3].max == hek_query[0]['obs_meanwavel'] * u.Unit(hek_query[0]['obs_wavelunit'])
-    assert vso_query[3].unit == u.Unit('Angstrom')
+    assert vso_query[3].min == hek_query[0]["obs_meanwavel"] * u.Unit(
+        hek_query[0]["obs_wavelunit"]
+    )
+    assert vso_query[3].max == hek_query[0]["obs_meanwavel"] * u.Unit(
+        hek_query[0]["obs_wavelunit"]
+    )
+    assert vso_query[3].unit == u.Unit("Angstrom")
 
 
+@pytest.mark.remote_data
 class TestH2VClient(object):
     """Tests the H2V class"""
-    # TODO
+
+    def test_members(self, h2v_client):
+        client = h2v_client
+        assert isinstance(client.hek_client, hek.HEKClient)
+        assert isinstance(client.vso_client, vso.VSOClient)
+        assert client.hek_results == ""
+        assert client.vso_results == []
+        assert client.num_of_records == 0
+
+    def test_translate_and_query(self, h2v_client, hek_client):
+        h = hek_client
+        h2v = h2v_client
+        q = h.search(
+            hek.attrs.Time(startTime, endTime), hek.attrs.EventType(eventType)
+        )
+        h2v_q = h2v.translate_and_query(q)
+
+        assert len(q) == len(h2v_q)
+        assert isinstance(h2v_q, list)
+        assert isinstance(h2v_q[0], vso.vso.QueryResponse)
+
+    def test_full_query(self, h2v_client, hek_client):
+        h2v = h2v_client
+        h = hek_client
+        h2v_q_1 = h2v.full_query(
+            (
+                hek.attrs.Time(startTime, endTime),
+                hek.attrs.EventType(eventType),
+            )
+        )
+
+        assert h2v.num_of_records == 2908
+        assert len(h2v.vso_results) == 19
+        assert len(h2v.hek_results) == 19
+
+        h2v._quick_clean()
+        q = h.search(
+            hek.attrs.Time(startTime, endTime), hek.attrs.EventType(eventType)
+        )
+        h2v_q_2 = h2v.translate_and_query(q)
+
+        assert len(h2v_q_1) == len(h2v_q_2)
+        assert len(h2v.hek_results) == len(q)
+        assert h2v.hek_results[0].get_voevent() == q[0].get_voevent()
+
+        for i in range(len(h2v_q_1)):
+            assert len(h2v_q_1[i]) == len(h2v_q_2[i])
+            if i != 2:
+                assert h2v_q_1[i].total_size() == h2v_q_2[i].total_size()
+
+    def test_quick_clean(self, h2v_client, hek_client):
+        h2v = h2v_client
+        h2v_q = h2v.full_query(
+            (
+                hek.attrs.Time(startTime, endTime),
+                hek.attrs.EventType(eventType),
+            )
+        )
+
+        assert h2v.num_of_records != 0
+        assert len(h2v.vso_results) != 0
+
+        h2v._quick_clean()
+
+        assert h2v.vso_results == []
+        assert h2v.num_of_records == 0
```

Comment:
Was this formatted by yapf?

Reply:
No, I gave this line space to separate other statements from assert ones.